### PR TITLE
Make kernel names shorter for name generator case

### PR DESCRIPTION
**Pull request by rarutyun**
Changes:

- Make kernel names shorter for name generator case. Add aliases for readability.
- Add comments for optional kernel name and for kernel name composer
- Remove `DefaultKernelNameFPGA`

### CODE REVIEW

File: include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h [hunk 1]
```diff
@@ -174,20 +174,34 @@ make_iter_mode(const _Iterator& __it) -> decltype(iter_mode<outMode>()(__it))
...
+class __pslk;
+
+template <typename... _Name>
+using __parallel_scan_local_kernel = __pslk<_Name...>;
+
+template <typename... _Name>
+class __psgrk;
+
+template <typename... _Name>
+using __parallel_scan_group_kernel = __psgrk<_Name...>;
```
Comment (MikeDvorskiy) [inline include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h:195]: I'm not sure about identificators as an abbreviations:
1) the abbreviations are applicable, but if it a kind of more or less standard term, like USM or, CPU
2) it  is not readable at all - "psgrk"...

Comment (rarutyun) [reply include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h:195]: The reason is to make kernel name shorter to reduce compilation time. What do you suggest?

For readability the aliases remain with the old names.

File: include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h [hunk 2]
```diff
@@ -174,20 +174,34 @@ make_iter_mode(const _Iterator& __it) -> decltype(iter_mode<outMode>()(__it))
 }
 
 // set of templated classes to name kernels
+// Shoten the names used in kernel generator to potentially improve compilation time
```
Comment (MikeDvorskiy) [inline include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h:177]: Shoten => Shorten
I propose to mention the all unreadable names  defined below in this description.

File: include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h (hunk 1 above)
Comment (rarutyun) [reply include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h]: Discussed offline. Make groups division clearer in the code. More information was added to the comments.

File: include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h (hunk 2 above)
Comment (rarutyun) [reply include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h]: Done. Comments rewritten

File: include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h [hunk 3]
```diff
@@ -157,6 +153,9 @@ struct __composite_kernel_name
 {
 };
 
+// Compose kernel name by transforming the constexpr string to the sequence of chars
+// and intitiate template with variadic non-type template parameters.
```
Comment (dsobolev-dev) [inline include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h:157]: intitiate -> instantiate

Comment (rarutyun) [reply include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h]: Done

**Comment by rarutyun** [comment]
CI is almost passed on the previous commit. The latest change is fix the typo in one word. So, there is no need to wait to CI completion to merge this pull request.

**Review by dsobolev-dev — APPROVED** [review]

**Pull request merged**